### ISSUE TITLE
JSPath modifies the input object when it contains an array, which is unexpected behaviour

### ISSUE THREAD

**Issue by alehmr** [dfilatov/jspath#33]
Please see attached image 

![image](https://cloud.githubusercontent.com/assets/1460893/6211312/9c455408-b5e9-11e4-8d44-9aed2f88d989.png)

Suite:

```javascript
var chai = require('chai');
var expect = chai.expect;
var jsPath = require("jspath");


describe('jspath', function () {

    it('array', function () {
        var obj = {
            a: [
                {b: 1, c: {b: 2}},
                {n: 2}
            ]
        };

        expect(obj.a).to.have.length(2);
        var b = jsPath.apply("..*.b", obj);
        expect(b).to.have.length(2);
        expect(obj.a).to.have.length(2, 'jspath does not modify the input: ' + JSON.stringify(obj.a, null, '\t'));
    });

});
```

**Comment by dfilatov** [comment]
It's definitely a bug, I'll investigate it asap.

**Comment by dfilatov** [comment]
fixed in 0.3.1

**Comment by dfilatov** [comment]
@alehmr Thanks for report!
